library/Icingadb/Model/Behavior/HasProblematicParent.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/* Icinga DB Web | (c) 2024 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Model\Behavior;
+
+use Icinga\Module\Icingadb\Model\DependencyEdge;
+use ipl\Orm\AliasedExpression;
+use ipl\Orm\ColumnDefinition;
+use ipl\Orm\Exception\InvalidColumnException;
+use ipl\Orm\Query;
+use ipl\Sql\Expression;
+use ipl\Stdlib\Filter;
+use ipl\Orm\Contract\RewriteColumnBehavior;
+use ipl\Orm\Contract\QueryAwareBehavior;
+
+/**
+ * Behavior to check if the service has a problematic parent
+ */
+class HasProblematicParent implements RewriteColumnBehavior, QueryAwareBehavior
+{
+    /** @var Query */
+    protected $query;
+
+    public function setQuery(Query $query): self
+    {
+        $this->query = $query;
+
+        return $this;
+    }
+
+    public function rewriteColumn($column, ?string $relation = null): ?AliasedExpression
+    {
+        if (! $this->isSelectableColumn($column)) {
+            return null;
+        }
+
+        $resolver = $this->query->getResolver();
+        if ($relation !== null) {
+            $serviceTableAlias = str_replace('.', '_', $relation);
+            $column = $serviceTableAlias . "_$column";
+        } else {
+            $serviceTableAlias = $resolver->getAlias($this->query->getModel());
+        }
+
+        $subQueryModel = new DependencyEdge();
+        $subQuery = (new Query())
+            ->setDb($this->query->getDb())
+            ->setModel($subQueryModel)
+            ->columns([new Expression('1')])
+            ->with([
+                'from',
+                'to',
+                'dependency'
+            ])
+            ->limit(1)
+            ->filter(Filter::any(
+                Filter::equal('to.redundancy_group.state.failed', 'y'),
+                Filter::equal('dependency.state.failed', 'y')
```

Comment:
I'm sure the dependency has failed, in case the redundancy group has. So the latter is redundant in the filter.

Reply:
You have a point.